servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/BufferStrategies.java

```diff
@@ -279,9 +287,10 @@ void beforeNewAccumulatorEmitted(final Accumulator<T, B> newAccumulator) {
 
         @Override
         public void accumulate(@Nullable final T item) {
-            size++;
+            size = addWithOverflowProtection(size, 1);
```

Comment:
In this case, if the `sizeThreshold == 1`, until the next `accumulate` is invoked. After thinking about it more, don't think that we need any protection because after `countThresholdBreached` is invoked, we do not expect new invocations of `accumulate` method. I will add an assertion, but may change it to `IllegalStateException` if that will make more sense.